src/ParseObject.js

```diff
@@ -1614,10 +1666,84 @@ class ParseObject {
     singleInstance = false;
     CoreManager.setObjectStateController(UniqueInstanceStateController);
   }
+
+  /**
+   * Stores the objects and every object they point to in the local datastore,
+   * recursively, using a default pin name: _default.
+   *
+   * @param {Array} objects A list of <code>Parse.Object</code>.
+   * @static
+   */
+  static pinAll(objects: Array<ParseObject>) {
+    const localDatastore = CoreManager.getLocalDatastore();
+    ParseObject.pinAllWithName(localDatastore.DEFAULT_PIN, objects);
+  }
+
+  /**
+   * Stores the objects and every object they point to in the local datastore, recursively.
+   *
+   * @param {String} name Name of Pin.
+   * @param {Array} objects A list of <code>Parse.Object</code>.
+   * @static
+   */
+  static pinAllWithName(name: string, objects: Array<ParseObject>) {
+    const localDatastore = CoreManager.getLocalDatastore();
+    for (const object of objects) {
+      localDatastore._handlePinWithName(name, object);
+    }
+  }
+
+  /**
+   * Removes the objects and every object they point to in the local datastore,
+   * recursively, using a default pin name: _default.
+   *
+   * @param {Array} objects A list of <code>Parse.Object</code>.
+   * @static
+   */
+  static unPinAll(objects: Array<ParseObject>) {
+    const localDatastore = CoreManager.getLocalDatastore();
+    ParseObject.unPinAllWithName(localDatastore.DEFAULT_PIN, objects);
+  }
+
+  /**
+   * Removes the objects and every object they point to in the local datastore, recursively.
+   *
+   * @param {String} name Name of Pin.
+   * @param {Array} objects A list of <code>Parse.Object</code>.
+   * @static
+   */
+  static unPinAllWithName(name: string, objects: Array<ParseObject>) {
+    const localDatastore = CoreManager.getLocalDatastore();
+    for (const object of objects) {
+      localDatastore._handleUnPinWithName(name, object);
+    }
+  }
+
+  /**
+   * Removes all objects in the local datastore using a default pin name: _default.
+   *
+   * @static
+   */
+  static unPinAllObjects() {
+    const localDatastore = CoreManager.getLocalDatastore();
+    localDatastore.unPinWithName(localDatastore.DEFAULT_PIN);
```

Comment:
we should hasve a method

```
getPinName(pinName: String?): string {
   if (!pinName) { return DEFAULT_PIN }
   return PIN_PREFIX + name;
}
```

This qwould void the probvlems of concatenating localDatastore.PIN_PREFIX + name or getting default :)